src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -71,18 +71,14 @@ internal class FindUsagesLSPContext : FindUsagesContext
             _position = position;
             _metadataAsSourceFileService = metadataAsSourceFileService;
             _workQueue = new AsyncBatchingWorkQueue<VSReferenceItem>(
-                TimeSpan.FromSeconds(0.5), ReportIfNotEmptyAsync, cancellationToken);
+                TimeSpan.FromSeconds(0.5), ReportReferencesAsync, cancellationToken);
 
             CancellationToken = cancellationToken;
         }
 
-        public override Task OnCompletedAsync()
-        {
-            // Upon completion, we wait an additional 0.5s (the time in between batches) to ensure
-            // that all results have been reported.
-            Thread.Sleep(500);
-            return Task.CompletedTask;
-        }
+        // After all definitions/references have been found, wait here until all results have been reported.
+        public override async Task OnCompletedAsync()
+            => await _workQueue.WaitUntilCurrentBatchCompletesAsync().ConfigureAwait(false);
```

Comment:
This def works.  But a simpler approach is even:

```suggestion
        public override Task OnCompletedAsync()
            => _workQueue.WaitUntilCurrentBatchCompletesAsync();
```